autoload/pandoc/completion.vim

```diff
@@ -37,19 +57,158 @@ function! pandoc#completion#Init() abort "{{{1
     endif
 endfunction
 
+"
+" Write a function that fills s:mylist with unused image file paths
+"
+fun! s:FetchImageNames() abort
+    "
+    " Add all image extensions to l:filelist
+    "
+    let l:filelist = []
+    call extend(l:filelist, globpath(s:figdir, '*.png', 1, 1))
+    call extend(l:filelist, globpath(s:figdir, '*.jpg', 1, 1))
+    call extend(l:filelist, globpath(s:figdir, '*.svg', 1, 1))
+    call extend(l:filelist, globpath(s:figdir, '*.gif', 1, 1))
+    call extend(l:filelist, globpath(s:figdir, '*.eps', 1, 1))
+    "
+    for l:file in l:filelist
+        if ! search(l:file, 'nw')
+            call add(s:mylist, {
+                        \ 'icase': 0,
+                        \ 'word': l:file,
+                        \ 'menu': '  [ins-Figure]',
+                        \ })
+        endif
+    endfor
+endfun
+
+"
+" Populate s:mylist with labels such as tbl:, eq:, fig: and tbl:
+"
+fun! s:FetchRefLabels() abort
+    let l:reflist = []
+    call execute(
```

Comment:
Maybe this could be done for all open buffers at least by `bufdo`? (...and at some point by all globbed markdown in the work dir?)

Reply:
This makes sense when you split your work among multiple files.I have made use of the bufdo feature in the most recent commit.

Reply:
Good job! It makes sense to restrict to only all those buffers that have `&l:filetype='pandoc'`.

There is also `argdo` after `argadd *.{pandoc,md}` (where the extensions are given by `ftdetect/pandoc.vim`). The argument list should be stored and restore.
Maybe optionally glob in the all sub directories, that is, `argadd **/*.{pandoc,md}`.

Reply:
Thinking about it, `**/*.{pandoc,md}` seems more reasonable by default. Since this could take time, better load it once initially and then refresh it by an `autocmd BufWrite`.

Reply:
Do you still wish to store and restore arglist ?

Reply:
Yes. Temporarily overriding the arglist is, not unusual in Vimscript, not the cleanest solution, but readily implemented once everything set up for the buffer list. Perhaps the `autocmd` update can be skipped for the time being: New items are in the buffer list anyway. Of course such a buffer can be closed, but whether the effort is worth it remains questionable.

Reply:
Added a variable called g:pandoc#completion#refsources that lets the user to choose to use all pandoc files from currently opened buffers, or to use all 'pandoc' files in the current directory (recursively).

Reply:
Very good. Thank you! Let's see if the maintainers @alerque and @fmoralesc deem it good to go?!

Reply:
Is there a way to avoid the flicker when the completion entry list is first populated?